sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/xml/StaxXmlGenerator.scala

```diff
@@ -83,21 +86,21 @@ private[xml] object StaxXmlGenerator {
 
     def writeElement(dt: DataType, v: Any, options: XmlOptions): Unit = (dt, v) match {
       case (_, null) | (NullType, _) => writer.writeCharacters(options.nullValue)
+      case (StringType, v: UTF8String) => writer.writeCharacters(v.toString)
       case (StringType, v: String) => writer.writeCharacters(v)
       case (TimestampType, v: Timestamp) =>
```

Comment:
I think you can remove this, and `case (DecimalType(), v: java.math.BigDecimal) => writer.writeCharacters(v.toString)`.

BTW, we would also add the type supports for `TimestampNTZType`, `YearMonthIntervalType` and `DayTimeIntervalType`. But they are orthogonal and can be done separately.

Reply:
JacksonParser supports DecimalType and I was planning to add support for DecimalType in a followup. Is that not required?